prober/main.go

```diff
@@ -0,0 +1,225 @@
+  provider: %s
+annotations:
+  traceID: %s
+contacts:
+  email:
+  - %s
+`, proberGCSNamePrefix, cfg.GCSBucketID, proberResourceProvider, traceID, proberResourceContact))
+
+	filepath := fmt.Sprintf("%s/%s-%s", cfg.ProberMappingServiceName, cfg.ProberMappingFilePrefix, timestamp)
+
+	if err := testhelper.UploadGCSFile(ctx, gcsClient, cfg.GCSBucketID, filepath, bytes.NewReader(data), getProberGCSMetadata()); err != nil {
+		return fmt.Errorf("failed to uploaded mapping object: %w", err)
+	}
+
+	queryString := fmt.Sprintf("SELECT data FROM `%s.%s.%s`", cfg.ProjectID, cfg.BigQueryDataSetID, cfg.MappingTableID)
+	queryString += ` WHERE JSON_VALUE(data.payload.annotations.traceID) = ?`
+
+	bqQuery := bqClient.Query(queryString)
+	bqQuery.Parameters = []bigquery.QueryParameter{{Value: traceID}}
+
+	gotBQEntry, err := getFirstMatchedBQEntry(ctx, bqQuery, cfg)
+	if err != nil {
+		return fmt.Errorf("failed to get match bigquery result: %w", err)
+	}
+
+	wantResourceMapping := &v1alpha1.ResourceMapping{
+		Resource: &v1alpha1.Resource{
+			Provider: proberResourceProvider,
+			Name:     fmt.Sprintf("%s/%s", proberGCSNamePrefix, cfg.GCSBucketID),
+		},
+		Contacts: &v1alpha1.Contacts{Email: []string{proberResourceContact}},
+	}
+
+	gotPmapEvent := &v1alpha1.PmapEvent{}
+	if err := protojson.Unmarshal([]byte(gotBQEntry), gotPmapEvent); err != nil {
+		return fmt.Errorf("failed to unmarshal bigquery result to pmapevent: %w", err)
+	}
+
+	resourceMapping := &v1alpha1.ResourceMapping{}
+	if err := gotPmapEvent.GetPayload().UnmarshalTo(resourceMapping); err != nil {
+		return fmt.Errorf("failed to unmarshal to resource mapping: %w", err)
+	}
+
+	cmpOpts := []cmp.Option{
+		protocmp.Transform(),
+		protocmp.IgnoreFields(&v1alpha1.ResourceMapping{}, "annotations"),
+	}
+
+	var diffErr error
+	if diff := cmp.Diff(wantResourceMapping, resourceMapping, cmpOpts...); diff != "" {
+		diffErr = errors.Join(diffErr, fmt.Errorf("resourcemapping(ignore annotation) unexpected diff (-want,+got):\n%s", diff))
+	}
+
+	wantGithubSource := &v1alpha1.GitHubSource{
+		RepoName:           proberGithubRepoValue,
+		Commit:             proberGithubCommitValue,
+		Workflow:           proberWorkflowValue,
+		WorkflowSha:        proberWorkflowShaValue,
+		WorkflowRunId:      proberWorkflowRunID,
+		WorkflowRunAttempt: 1,
+	}
+
+	if diff := cmp.Diff(wantGithubSource, gotPmapEvent.GetGithubSource(), protocmp.Transform()); diff != "" {
+		diffErr = errors.Join(diffErr, fmt.Errorf("githubSource unexpected diff (-want, +got):\n%s", diff))
+	}
+
+	// check CAIS annotation exist to make sure CAIS is working
+	if _, ok := resourceMapping.GetAnnotations().GetFields()[v1alpha1.AnnotationKeyAssetInfo].GetStructValue().AsMap()["ancestors"]; !ok {
+		diffErr = errors.Join(diffErr, fmt.Errorf("ancestors is blank in resourcemapping.annotations"))
+	}
+	if _, ok := resourceMapping.GetAnnotations().GetFields()[v1alpha1.AnnotationKeyAssetInfo].GetStructValue().AsMap()["iamPolicies"]; !ok {
```

Comment:
Do we have any iamPolicies set in the gcs bucket?

Reply:
Yes, we will be using the gcs bucket where we upload files to do the resource mapping. From https://github.com/abcxyz/pmap/blob/main/terraform/modules/common/main.tf#L167, this bucket is guarenteed to have iamPolicys.